WIP Fix failing Keystore Passphrase test for feature branch

williamrandolph opened the pull request:
On the cleaned-up feature branch for the password-protected keystore, there are two problems with one of our packaging tests: first, it fails occasionally, and second, it doesn't clean up after itself when it does, so a bunch of later tests also fail.

It appears that the issue is that some systems store journald logs in `/run/log/journal` while Fedora and Ubuntu store them in `/var/log/journal`, so our method that clears out the journald logs shoudl clear both locations.

Fixes #50079

williamrandolph committed:
Remove Ignore annotation for broken test

williamrandolph committed:
Use try/finally and re-remove distro conditional

One problem with the passphrase-from-file tests, as written, is that
they would leave a SystemD environment variable set when they failed,
and this setting would cause elasticsearch startup to fail for other
tests as well. By using a try-finally, I hope that these tests will fail
more gracefully.

I also found that the "packagingConditional" pattern was still in use on
this branch despite being removed in master, so I'm removing it again
here.

elasticmachine commented:
Pinging @elastic/es-core-infra (:Core/Infra/Packaging)

williamrandolph committed:
Merge branch 'feature-pwd-protected-keystore-2' into keystore-password-from-file

williamrandolph committed:
Update keystore tests for merge from master

We need to use a new pattern for CLI tools. There's also some checkstyle
cleanup here.

rjernst commented on qa/os/src/test/java/org/elasticsearch/packaging/test/KeystoreManagementTests.java:
```diff
@@ -93,7 +93,7 @@ public void test20CreateKeystoreManually() throws Exception {
             ? "sudo -u " + ARCHIVE_OWNER + " "
             : "";
-        Shell.Result r = sh.run(possibleSudo + bin.elasticsearchKeystore + " list");
+        Shell.Result r = sh.run(possibleSudo + bin.keystoreTool + " list");
```
You can use keystoreTool.run like above now right?

rjernst commented on qa/os/src/test/java/org/elasticsearch/packaging/test/KeystoreManagementTests.java:
```diff
@@ -192,87 +192,74 @@ public void test50KeystorePasswordFromFile() throws Exception {
+            switch (distribution.packaging) {
+                case TAR:
+                case ZIP:
+                    sh.run("sudo -u " + ARCHIVE_OWNER + " " + bin.keystoreTool + " create");
```
I thought I made run so it will do the sudo if necessary, so this should not be necessary and you can run `bin.keystoreTool.run("create")`, in which case I'm not sure we need any of these platform specific blocks?

williamrandolph replied:
Yep, you're right! I though of that while bringing in the master branch, and now I've followed up on it.

rjernst reviewed (approved):
LGTM. I think there are several places that can use the `run()` method on the keystore tool Executable.

williamrandolph committed:
Use new tool patterns in os tests

williamrandolph committed:
Use new tool patterns in two other os tests

williamrandolph committed:
Clear persistent journald directory

It appears that our Fedora and Ubuntu environments may be configured to
store journald information under /var rather than under /run, so that it
will persist between boots. Our destructive tests that read from the
journal need to account for this in order to avoid trying to limit the
output we check in tests.

williamrandolph committed:
Merge branch 'feature-pwd-protected-keystore-2' into keystore-password-from-file

williamrandolph commented on qa/os/src/test/java/org/elasticsearch/packaging/util/Packages.java:
```diff
@@ -286,7 +286,7 @@ private static void verifyDefaultInstallation(Installation es) {
     public static void clearJournal(Shell sh) {
         if (isSystemd()) {
-            sh.run("rm -rf /run/log/journal/*");
+            sh.run("rm -rf /run/log/journal/* /var/log/journal/*");
```
@rjernst All refactoring aside, here's the one-line fix for the test failure. Does this look OK to you?

rjernst replied:
Yes that seems fine.
